public/app/features/alerting/unified/components/contact-points/useNotificationTemplates.ts

```diff
@@ -0,0 +1,138 @@
+import { produce } from 'immer';
+
+import { ActiveTab as ContactPointsActiveTabs } from 'app/features/alerting/unified/components/contact-points/ContactPoints';
+import { useDispatch } from 'app/types';
+
+import { AlertManagerCortexConfig } from '../../../../../plugins/datasource/alertmanager/types';
+import { alertmanagerApi } from '../../api/alertmanagerApi';
+import { updateAlertManagerConfigAction } from '../../state/actions';
+import { ensureDefine } from '../../utils/templates';
+import { TemplateFormValues } from '../receivers/TemplateForm';
+
+// TODO Check the issue with useAlertmanagerConfig hook
+export function useNotificationTemplates(amSourceName?: string) {
+  const templatesRequestState = alertmanagerApi.endpoints.getAlertmanagerConfiguration.useQuery(amSourceName || '', {
+    skip: !amSourceName,
+    selectFromResult: (state) => ({
+      ...state,
+      data: state.data ? amConfigToTemplates(state.data) : undefined,
+      currentData: state.currentData ? amConfigToTemplates(state.currentData) : undefined,
+    }),
+  });
+
+  return templatesRequestState;
+}
+
+export interface NotificationTemplate {
+  name: string;
+  template: string;
+  provenance?: string;
+}
+
+function amConfigToTemplates(config: AlertManagerCortexConfig): NotificationTemplate[] {
+  return Object.entries(config.template_files).map(([name, template]) => ({
+    name,
+    template,
+    provenance: (config.template_file_provenances ?? {})[name],
+  }));
+}
+
+export function useCreateNotificationTemplate(amSourceName?: string) {
```

Comment:
Why the `amSourceName` is optional?

Reply:
Because the `useAlertmanager` hook returns selectedAlertmanager as `string | undefined`. I could just pass an empty string but I'm not sure if it's a better approach 🤔

Reply:
maybe we could add some early returns when `useAlertmanager` returns undefined so we don't render anything else?

Reply:
Unfortunately we cannot do early returns while using hooks :/ Hooks cannot be called conditionally, so early returns are not an option 😞